src/ServerRequestFactory.php

```diff
@@ -464,9 +468,8 @@ private static function createUploadedFileFromSpec(array $value)
      * @param array $files
      * @return UploadedFileInterface[]
      */
-    private static function normalizeNestedFileSpec(array $files)
+    private static function normalizeNestedFileSpec(array $files = [])
     {
-        $files = [];
```

Comment:
Actually, I think you've spotted an error; the value shouldn't be reset at the start of the method!